Debian specific build system options
This patch refreshes build system options in one place.
Formerly these were split amongst config_compilers_gcc295.mak.diff
and config_compilers_gcc_defaults.mak.diff. Since gcc 2.95 is long
obsolete we do this in one place now.
Author: Peter Drysdale <drysdalepete@gmail.com>

--- speech-tools-2.1~release.orig/config/compilers/gcc_defaults.mak
+++ speech-tools-2.1~release/config/compilers/gcc_defaults.mak
@@ -39,7 +39,7 @@
  ###########################################################################
 
 CC=gcc
-CXX=gcc
+CXX=g++
 ifeq ($(OSTYPE),Darwin)
 CXX=g++
 endif
@@ -48,14 +48,14 @@ COMPILER_DESC=FSF gcc
 COMPILER_VERSION_COMMAND=$(CXX) -v 2>&1 | tail -1 | sed -e 's/^....//'
 
 CFLAGS  = $(GCC_SYSTEM_OPTIONS) $(CC_OTHER_FLAGS)
-CXXFLAGS  =  $(GCC_SYSTEM_OPTIONS) -fno-implicit-templates $(CC_OTHER_FLAGS)
+CXXFLAGS  =  $(GCC_SYSTEM_OPTIONS) $(CC_OTHER_FLAGS)
 
 DEBUG_CCFLAGS   = -g
 DEBUG_CXXFLAGS  = -g
 DEBUG_LINKFLAGS = -g
 
 WARN_CCFLAGS   = -Wall
-WARN_CXXFLAGS  = -Wall
+WARN_CXXFLAGS  = -Wall -Wno-non-template-friend
 WARN_LINKFLAGS = -Wall
 
 VERBOSE_CCFLAGS   =
@@ -81,7 +81,7 @@ SHARED_CXXFLAGS  = -fPIC
 SHARED_LINKFLAGS = 
 
 ifndef GCC_MAKE_SHARED_LIB
-    MAKE_SHARED_LIB = $(CXX) -shared -fno-shared-data -o XXX
+    MAKE_SHARED_LIB = $(CXX) -shared -o XXX -Wl,-soname -Wl,YYY
 else
     MAKE_SHARED_LIB = $(GCC_MAKE_SHARED_LIB)
 endif
@@ -98,9 +98,10 @@ TEMPLATE_ARGS =
 ## The -lgcc here is redundant - gcc does this anyway - but it
 ## helps java know what needs to be loaded.
 
-COMPILERLIBS= $(COMPILER_LIBS_DIR:%=-L%) -lstdc++ -lgcc
+COMPILERLIBS= $(COMPILER_LIBS_DIR:%=-L%)
 
 ## special ways of doing things, blank means default
+STATIC_LINKFLAGS = -Dstatic
 
 MAKE_DEPEND_C = $(CC) -MM $(INCLUDES) $(TEMPLATES) $(TEMPLATE_SPECIFIC)
 MAKE_DEPEND_CXX = $(CC) -MM $(INCLUDES) $(TEMPLATES) $(TEMPLATE_SPECIFIC)
